### ISSUE TITLE
Support use cases that need to target a specific number of registers.

### ISSUE THREAD

**Issue by bjacob** [WebAssembly/simd#225]
This is open-ended. The problem is that many key use cases, such as matrix multiplication kernels, need to know a number of SIMD vector registers that they can count on using. In practice, the number of available architecture registers tend to be just large enough to hit peak performance, so matrix multiplication kernels tend to use all available registers. Here is an [example](https://github.com/google/ruy/blob/808ff748e0c7dc746a413fe45fa022d63e6253e8/ruy/kernel_arm64.cc#L113-L135).

**Comment by tlively** [comment]
> 1. Can the programmer query the number of architecture registers?

No, exposing underlying architectural details would introduce platform-specific behavior and violate WebAssembly's determinism. Although this kind of nondeterminism might be considered for a future proposal, it is out of scope for this SIMD proposal.

> 2. Can the programmer make assumptions about the correspondence between the number of SIMD vector variables used in a part of the program, and the register usage of the generated code?

No, different engines may make different register allocation decisions and may optimize or otherwise transform the code however they deem fit, so programmers should not be making these sorts of assumptions. It may be possible to make assumptions about codegen for a particular engine, but it should not be assumed that those assumptions will generalize to other engines.

The low-level, portable SIMD instructions in this proposal have proven to be useful for a wide variety of workloads, but we are aware that there are also many workloads that depend on non-portable instructions. Keep an eye out for future proposals meant to address this problem.
